File: server/router/image/image.go
package image

import (
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

type RouterGroup struct {
	ImageRouter
}

type ImageRouter struct{}

func (r *ImageRouter) InitImageRouter(Router *gin.RouterGroup) {
	imageRouter := Router.Group("image").Use(middleware.OperationRecord())
	imageWithNotRecorderRouter := Router.Group("image")
	imageApi := imageApiGroup.ImageApi
	{
		imageRouter.POST("add", imageApi.CreateImage)    // 创建镜像
		imageRouter.POST("update", imageApi.UpdateImage) // 更新镜像
		imageRouter.POST("delete", imageApi.DeleteImage) // 删除镜像
	}
	{
		imageWithNotRecorderRouter.GET("list", imageApi.GetImageList) // 获取镜像列表
	}
}

File: web/src/api/image.ts
import service from '@/utils/request';

// 获取镜像列表
export const getImageList = (params?: any) => {
    return service({
        url: '/api/v1/image/list',
        method: 'get',
        params
    });
};

// 获取镜像详情
export const getImageDetail = (id: number | string) => {
    return service({
        url: '/api/v1/image/detail',
        method: 'get',
        params: { id }
    });
};

// 创建镜像
export const createImage = (data: any) => {
    return service({
        url: '/api/v1/image/add',
        method: 'post',
        data
    });
};

// 更新镜像
export const updateImage = (data: any) => {
    return service({
        url: '/api/v1/image/update',
        method: 'post',
        data
    });
};

// 删除镜像
export const deleteImage = (data: any) => {
    return service({
        url: '/api/v1/image/delete',
        method: 'post',
        data
    });
};
